### PR TITLE
fix: git account enable/disable switch doesn't work

### PR DESCRIPTION
**Pull request by rish2320**
# Description
Global config > Git account enable/disable switch doesn't work

Fixes #[1213](https://github.com/devtron-labs/devtron/issues/1213)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Tested all the cases manually.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by milindk8 — APPROVED** [review]

File: src/components/gitProvider/GitProvider.tsx [hunk 1]
```diff
@@ -201,21 +202,20 @@ function CollapsedList({
...
-        if (!collapsed) return
         async function update() {
             let payload = {
                 id: id || 0,
                 name,
                 url,
                 authMode,
                 active: enabled,
-                gitHostId,
+                gitHostId: (Number)(gitHostId),
```
Comment (vivek-devtron) [inline src/components/gitProvider/GitProvider.tsx:218]: we can write this as +gitHostId

File: src/components/gitProvider/GitProvider.tsx [hunk 2]
```diff
@@ -284,11 +288,11 @@ function CollapsedList({
                             placement="bottom"
                             content={enabled ? 'Disable git account' : 'Enable git account'}
                         >
-                            <span style={{ marginLeft: 'auto' }} data-testid={`${name}-toggle-button`}>
+                            <span onClick={stopPropagation} style={{ marginLeft: 'auto' }} data-testid={`${name}-toggle-button`}>
                                 {loading ? (
                                     <Progressing />
                                 ) : (
-                                    <List.Toggle onSelect={(en) => toggleEnabled(en)} enabled={enabled} />
+                                    <List.Toggle onSelect={(en) => setToggleEnabled(en)} enabled={enabled} />
```
Comment (vivek-devtron) [inline src/components/gitProvider/GitProvider.tsx:295]: can we directly write onSelect={toggleEnabled}?

Comment (rish2320) [reply src/components/gitProvider/GitProvider.tsx]: done

File: src/components/gitProvider/GitProvider.tsx (hunk 1 above)
Comment (rish2320) [reply src/components/gitProvider/GitProvider.tsx]: done

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**